contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -193,20 +226,14 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     bytes calldata
   ) external view override whenNotPaused returns (bool upkeepNeeded, bytes memory performData) {
     address[] memory needsFunding = sampleUnderfundedAddresses();
-    uint256 numCanFund = LINK_TOKEN.balanceOf(address(this)) / s_topUpAmount;
-    if (numCanFund < needsFunding.length) {
-      assembly {
-        mstore(needsFunding, numCanFund) // resize
-      }
-    }
     upkeepNeeded = needsFunding.length > 0;
     performData = abi.encode(needsFunding);
     return (upkeepNeeded, performData);
   }
 
   /// @notice Called by the keeper to send funds to underfunded addresses.
   /// @param performData the abi encoded list of addresses to fund
-  function performUpkeep(bytes calldata performData) external override {
+  function performUpkeep(bytes calldata performData) external override whenNotPaused {
```

Comment:
whenNotPaused is redundant, because `performUpkeep()` calls `topUp()`, which has the same modifier